presto-spi/src/main/java/io/prestosql/spi/security/CertificateAuthenticator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.security;
+
+import java.security.Principal;
+import java.security.cert.X509Certificate;
+
+public interface CertificateAuthenticator
+{
+    /**
+     * Authorize using the certificate's principal
+     *
+     * @return true when authorized
+     * @throws AccessDeniedException if not allowed
+     */
+    Principal authenticate(X509Certificate[] certs);
```

Comment:
Let's change this to `List<X509Certificate>` since lists are preferred to arrays (see https://www.informit.com/articles/article.aspx?p=2861454&seqNum=3)

Also rename to `certificates` since we avoid abbreviations when naming things.

Reply:
Add Javadoc explaining this
```java
/**
 * @param certificates This client certificate chain, in ascending order of trust.
 * The first certificate in the chain is the one set by the client, the next is the
 * one used to authenticate the first, and so on.
 * @return the authenticated entity
 * @throws AccessDeniedException if not allowed
 */